lib/pages/class_feedback/view/class_feedback_view.dart

```diff
@@ -0,0 +1,276 @@
+      ];
+      for (final question
+          in feedbackQuestions.values.where((q) => q.category == category)) {
+        categoryChildren.add(FeedbackQuestionForm(
+            question: question, answerValues: answerValues, formKey: formKey));
+      }
+      children.add(
+        Column(
+          children: [
+            Card(
+              child: Padding(
+                padding: const EdgeInsets.all(15),
+                child: Column(
+                  children: categoryChildren,
+                ),
+              ),
+            ),
+            const SizedBox(height: 24),
+          ],
+        ),
+      );
+    }
+
+    return AppScaffold(
+      title: Text(S.current.navigationClassFeedback),
+      actions: [_submitButton()],
+      body: ListView(
+        children: <Widget>[
+          Center(
+            child: Padding(
+              padding: const EdgeInsets.only(left: 25, top: 10),
+              child: IconText(
+                icon: Icons.info_outline,
+                text: S.current.infoFormAnonymous,
+                style: Theme.of(context).textTheme.bodyText1,
+              ),
+            ),
+          ),
+          Padding(
+            padding: const EdgeInsets.all(10),
+            child: Form(
+              key: formKey,
+              child: Column(mainAxisSize: MainAxisSize.min, children: children),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  AppScaffoldAction _submitButton() => AppScaffoldAction(
+        text: S.current.buttonSend,
+        onPressed: () async {
+          if (!formKey.currentState.validate()) return;
+
+          if (!agreedToResponsibilities) {
+            AppToast.show(
+                '${S.current.warningAgreeTo}${S.current.labelFeedbackPolicy}.');
+            return;
+          }
+
+          setState(() {
+            formKey.currentState.save();
+          });
+
+          bool res1, res2;
+          final authProvider =
+              Provider.of<AuthProvider>(context, listen: false);
+          final String uid = authProvider.uid;
+          for (var i = 0; i < feedbackQuestions.length; i++) {
+            res1 = false;
+
+            final response = FeedbackQuestionAnswer(
+              assistant: selectedAssistant,
+              teacher: selectedTeacher,
+              className: classController.text,
+              questionNumber: i.toString(),
+              questionAnswer: feedbackQuestions[i.toString()].answer,
+            );
+
+            res1 = await Provider.of<FeedbackProvider>(context, listen: false)
+                .addResponse(response);
+            if (!res1) break;
+          }
+
+          res2 = await Provider.of<FeedbackProvider>(context, listen: false)
+              .setUserClassFeedback(classController.text, uid);
+
+          if (res1 && res2) {
+            Navigator.of(context).pop();
+            AppToast.show(S.current.messageFeedbackHasBeenSent);
+          }
+        },
```

Comment:
This is the business logic that shouldn't be in the UI (see my comment above). Move this to a method (`submitForm`?) inside the provider, it will look a lot better too (and I'm sure you can do better at naming than `res1` and `res2`).